Gem/Code/Source/SampleComponentManager.cpp

```diff
@@ -1490,7 +1503,10 @@ namespace AtomSampleViewer
         pipelineDesc.m_name = "RPISamplePipeline";
         pipelineDesc.m_rootPassTemplate = GetRootPassTemplateName();
         pipelineDesc.m_mainViewTagName = "MainCamera";
-        pipelineDesc.m_renderSettings.m_multisampleState.m_samples = GutNumMSAASamples();
+
+        // set pipeline MSAA samples
+        AZ_Assert(m_numMSAASamples >= 1 && m_numMSAASamples <= 8, "Invalid MSAA sample setting");
```

Comment:
Do we need to check for specific numbers here... like only 1,2,4 and 8 are valid right?

Reply:
Good idea, added.